scripts/keeper.ts

```diff
@@ -1,221 +1,171 @@
   );
-  await provider.waitForTransaction(initializeTxHash);
-  const account = new starknet.Account(provider, accountContract.address, starkKeyPair);
-  console.log(account.address);
-}
-
-async function flush(targetDomain: string) {
-  const l1Signer = getL1Signer(NETWORK);
-  const l2Signer = getL2Signer(NETWORK);
-
-  const l1WormholeGatewayAddress = getRequiredEnv(`${NETWORK}_L1_DAI_WORMHOLE_GATEWAY_ADDRESS`);
-  const l1WormholeGateway: L1DAIWormholeGateway = (await getL1ContractAt(l1Signer, "L1DAIWormholeGateway", l1WormholeGatewayAddress) as L1DAIWormholeGateway);
-
-  const l2WormholeGatewayAddress = getRequiredEnv(`${NETWORK}_L2_DAI_WORMHOLE_GATEWAY_ADDRESS`);
-  const l2WormholeGateway: l2_dai_wormhole_gateway = (await getL2ContractAt(l2Signer, "l2_dai_wormhole_gateway", l2WormholeGatewayAddress) as l2_dai_wormhole_gateway);
-
-  const wormholeJoinInterface = new ethers.utils.Interface([
-    "event Settle(bytes32 indexed sourceDomain, uint256 batchedDaiToFlush)",
-  ]);
-  const l1WormholeJoinAddress = getRequiredEnv(`${NETWORK}_L1_WORMHOLE_JOIN_ADDRESS`);
-  const l1WormholeJoin = new ethers.Contract(l1WormholeJoinAddress, wormholeJoinInterface, l1Signer);
-  const filter = l1WormholeJoin.filters.Settle(l1String(SOURCE_DOMAIN));
-  const events = await l1WormholeJoin.queryFilter(filter);
-  const recentEvent = events[events.length - 1];
-
-  const encodedDomain = l2String(targetDomain);
-  const [daiToFlushSplit] = await l2WormholeGateway.batched_dai_to_flush(encodedDomain);
-  const daiToFlush = toUint(daiToFlushSplit);
-  console.log(`DAI to flush: ${daiToFlush}`);
-
-  // check last settle event and amount to flush
-  const currentBlock = await l1Signer.provider.getBlockNumber();
-  if (daiToFlush > 0 && (!recentEvent || recentEvent.blockNumber > currentBlock + FLUSH_DELAY)) {
-    console.log("Sending `flush` transaction");
-    const tx = await l2WormholeGateway.flush(encodedDomain, { maxFee: "0" });
-    const res = await tx.wait();
-    console.log("Success");
+  const recentEvent = settleEvents[settleEvents.length - 1];
+  if (recentEvent) {
+    const block = await recentEvent.getBlock();
+    return block.timestamp;
+  } else {
+    return 0;
   }
 }
 
-async function finalizeFlush(targetDomain: string) {
-  const l1Signer = getL1Signer(NETWORK);
-  const l2Signer = getL2Signer(NETWORK);
-
-  const l1WormholeGatewayAddress = getRequiredEnv(`${NETWORK}_L1_DAI_WORMHOLE_GATEWAY_ADDRESS`);
-  const l2WormholeGatewayAddress = getRequiredEnv(`${NETWORK}_L2_DAI_WORMHOLE_GATEWAY_ADDRESS`);
-  const starknetAddress = getRequiredEnv(`${NETWORK}_STARKNET_ADDRESS`);
-  const l1WormholeGateway: L1DAIWormholeGateway = (await getL1ContractAt(l1Signer, "L1DAIWormholeGateway", l1WormholeGatewayAddress) as L1DAIWormholeGateway);
-  const l2WormholeGateway: l2_dai_wormhole_gateway = (await getL2ContractAt(l2Signer, "l2_dai_wormhole_gateway", l2WormholeGatewayAddress) as l2_dai_wormhole_gateway);
+async function flushesToBeFinalized(
```

Comment:
Can't we simply query `LogMessageToL1` with `fromBlock` set to either: `Date.now() - 10 * config.flushDelay`, or if there are no results preconfigured `startBlock`?
Btw: I would make `flushDelay multiplier` and `startingBlock` a .env parameters.